### PR TITLE
Enforce abstract classes can not be instantiated

### LINKED ISSUE #319: Tests for serialization and deserialization reference the abstract WiseAgent class
We should update the YAML files used in these tests and the objects referenced in the tests to reference the actual WiseAgent implementations instead.

### LINKED ISSUE #328: Prevent abstract class to be initialized
See #319

### PR DESCRIPTION
**Pull request by kabir**
Fixes #319 
FIxes #328

### CODE REVIEW

File: src/wiseagents/vectordb/wise_agent_vector_db.py [hunk 1]
```diff
@@ -24,9 +26,12 @@ class Document(BaseModel):
 
 class WiseAgentVectorDB(yaml.YAMLObject):
     """Abstract class to define the interface for a WiseAgentVectorDB."""
-    yaml_tag = u'!WiseAgentVectorDB'
     yaml_loader = WiseAgentsLoader
 
+    def __init__(self):
```
Comment (fjuma) [inline src/wiseagents/vectordb/wise_agent_vector_db.py:31]: Shouldn't an init method like this also be added for wise_agent_graph_db.py?

File: src/wiseagents/core.py [hunk 2]
```diff
@@ -27,7 +29,7 @@ class WiseAgentCollaborationType(Enum):
 
 
 class WiseAgentTool(yaml.YAMLObject):
-    ''' A WiseAgentTool is an abstract class that represents a tool that can be used by an agent to perform a specific task.'''
+    ''' WiseAgentTool represents a tool that can be used by an agent to perform a specific task.'''
```
Comment (fjuma) [inline src/wiseagents/core.py:32]: I'll let @maeste comment on what was intended for this one.

**Review by fjuma — CHANGES_REQUESTED** [review]

File: src/wiseagents/core.py (hunk 2 above)
Comment (kabir) [reply src/wiseagents/core.py:32]: FWIW I could not see anything extending this

File: src/wiseagents/vectordb/wise_agent_vector_db.py (hunk 1 above)
Comment (kabir) [reply src/wiseagents/vectordb/wise_agent_vector_db.py:31]: Yeah, I missed it - added now

File: src/wiseagents/core.py (hunk 2 above)
Comment (maeste) [reply src/wiseagents/core.py:32]: I'm good with this change

**Review by fjuma — APPROVED** [review]
Thanks @kabir!

**Pull request merged**